src/backend/commands/session/join_player.cpp

```diff
@@ -0,0 +1,96 @@
+#include "backend/command.hpp"
+#include "network/CNetworkPlayerMgr.hpp"
+#include "pointers.hpp"
+#include "services/players/player_service.hpp"
+#include "socialclub/FriendRegistry.hpp"
+#include "services/player_database/player_database_service.hpp"
+#include "services/api/api_service.hpp"
+#include "util/session.hpp"
+
+namespace big
+{
+	class join_player : command
+	{
+		using command::command;
+
+		virtual std::optional<std::vector<std::string>> get_argument_suggestions(int arg) override
+		{
+			if (arg == 1)
+			{
+				std::vector<std::string> suggestions;
+				for (size_t i = 0; i < g_pointers->m_gta.m_friend_registry->m_friend_count; i++)
+				{
+					auto f = g_pointers->m_gta.m_friend_registry->get(i);
+					if (f && f->m_friend_state & (1 << 0 | 1 << 1)) // Check if online and playing same game
+					{
+						suggestions.push_back(f->m_name);
+					}
+				}
+
+				for (const auto& player : g_player_database_service->get_sorted_players() | std::ranges::views::keys)
+					suggestions.push_back(player);
+
+				return suggestions;
+			}
+
+			return std::nullopt;
+		}
+
+		virtual std::optional<command_arguments> parse_args(const std::vector<std::string>& args, const std::shared_ptr<command_context> ctx) override
+		{
+			command_arguments result(1);
+
+			uint64_t rid = 0;
+			// Check if the player is a friend
+			for (size_t i = 0; i < g_pointers->m_gta.m_friend_registry->m_friend_count; i++)
+			{
+				auto f = g_pointers->m_gta.m_friend_registry->get(i);
+				if (f && f->m_name == args[0])
+				{
+					rid = f->m_rockstar_id;
+					break;
+				}
+			}
+
+			// If not a friend, check if the player is in the database
+			if (rid == 0)
+			{
+				for (const auto& [name, player] : g_player_database_service->get_sorted_players())
+					if (name == args[0])
+						rid = player->rockstar_id;
+			}
+
+			// If the player is not a friend or in the database, fetch rid from the API
+			if (rid == 0)
+			{
+				auto fetched = g_api_service->get_rid_from_username(args[0], rid);
```

Comment:
I know we coded this together, however this will cause a deadlock until the Rockstar API returns some data.

If we want to make this work we'll have to make it so suggestions can be given in an asynchronous way.
Alternatively if we were to support std::string (mainly making sure that the string survives longer than the argument creation context), then we could pass the string to the execute method.

Or as a botch fix we can make a member variable to pass data from the parse method to the execute method.